src/compiler/checker.ts

```diff
@@ -1192,6 +1195,40 @@ namespace ts {
                     return false;
                 });
             }
+
+            function isPropertyImmediatelyReferencedWithinDeclaration(declaration: PropertyDeclaration, usage: Node) {
+                // always legal if usage is after declaration
+                if (usage.end > declaration.end) {
+                    return true;

```

Comment:
The return value is backward from the name. Probably best to flip the value, not the name, because it's a pretty good name.